test/core/validate-graph.test.ts

```diff
@@ -0,0 +1,63 @@
+import { validatePackageURL } from '../../src/core/validate-graph';
+
+describe('validatePackageURL', () => {
+  describe('deb package type tests', () => {
+    it.each([
+      [
+        'both package and purl names contain a source',
```

Comment:
No, the purl name can't and shouldn't contain the source - that's an invalid deb-purl that you're describing here.

Reply:
Fixed!